### PR TITLE
Publish docker images to DockerHub

### LINKED ISSUE #783: Wrong READMEs in github packages
GitHub Packages uses the README from the root of the repository for all packages in the repository. If you want to have a specific README for a package in a monorepo, you can use a workaround.

The workaround is to temporarily move to the package directory, copy the package-specific README to the root, publish the package, and then move the README back. Here's how you can do it:

```bash
# Move to the package directory
cd package

# Copy the package README to the root
cp README.md ../

# Publish the package
npm publish

# Restore the root README back
mv README.md packages/my-package/
```

This will ensure that the package-specific README is used when publishing the package to GitHub Packages. Remember to replace `packages/my-package` with the actual path to your package directory.

courtesy: Github co-pilot

### PR DESCRIPTION
**Pull request by aryanpingle**
Intended to fix #783. 

Overview:
- Uses DockerHub as the container registry for the software's Docker images instead of GitHub Packages. (Note: the previously published GitHub packages will still show up under the `packages` tab of the sidebar)
- The DockerHub username must be a repository/environment variable called `DOCKERHUB_USERNAME`
- The DockerHub token must be a repository secret called `DOCKERHUB_TOKEN`
- Adds new markdown files to serve as DockerHub READMEs for the `client` and `libms` images

### CODE REVIEW

**Comment by aryanpingle** [comment]
@prasadtalasila I created `servers/lib/DOCKER.md` to match the conventions of `client/DOCKER.md`, is there anything in it that should be removed/edited to make it suitable for the Docker Image?

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -3,10 +3,10 @@ name: Build and Push Docker Image
...
   workflow_call:
     inputs:
-      registry:
-        required: false
-        type: string
-        default: "ghcr.io"
+      # registry:
+      #   required: false
+      #   type: string
+      #   default: "ghcr.io"
```
Comment (aryanpingle) [inline .github/workflows/docker.yml:9]: Removed this because we'll be using the default DockerHub registry.

**Comment by aryanpingle** [comment]
Additionally, many files reference the Docker Images on GitHub packages. Should we remove/edit those sections?

**Comment by prasadtalasila** [comment]
@aryanpingle we need to publish to both github and docker hub. Does the PR already support it?

File: .github/workflows/docker.yml (hunk 1 above)
Comment (prasadtalasila) [reply .github/workflows/docker.yml:9]: We need to publish on both github nd docker hub.

**Comment by aryanpingle** [comment]
Ah no, only to DockerHub. I'll add GH package support back.

**Comment by aryanpingle** [comment]
@prasadtalasila I've added back the GitHub Packages functionality.

Some notes:

- I've split the GitHub Packages and DockerHub workflows into separate files, and called both in separate jobs. I feel that calling both separately offers more control over the workflows, but I could put them in a combined `docker.yml` file and simply call that instead.
- The `check_version` step incorrectly left out the user name, which would result in it always assuming the version didn't exist. I've included the fix for it in [this commit](https://github.com/INTO-CPS-Association/DTaaS/pull/958/commits/f4e557cef85bcb66c68df25eaf32d65d3b2b76e3).
- In `lib-ms.yml` the `publish-docker-image` job did not include the push+branch condition, was this intentional?

**Comment by prasadtalasila** [comment]
@aryanpingle 
Please use these two edited files.
[Client-DOCKER.md](https://github.com/user-attachments/files/17310371/Client-DOCKER.md)
[LIB-DOCKER.md](https://github.com/user-attachments/files/17310374/LIB-DOCKER.md)

File: .github/workflows/docker-dockerhub.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        type: string
+        default: "latest"
+      dockerfile:
+        required: true
+        type: string
+      readme-file:
+        required: true
+        type: string
+    secrets:
+      DOCKERHUB_TOKEN:
```
[the remark this answers is not in the document]
Comment (aryanpingle) [reply .github/workflows/docker-dockerhub.yml:27]: Made it a repository secret instead

**Pull request merged**